x/marker/client/cli/tx.go

```diff
@@ -539,12 +546,135 @@ func GetNewTransferCmd() *cobra.Command {
 			if len(coins) != 1 {
 				return sdkErrors.Wrapf(sdkErrors.ErrInvalidCoins, "invalid coin %s", args[2])
 			}
+
+			authzGranter, err := cmd.Flags().GetString(FlagGranter)
+			if err != nil {
+				return sdkErrors.Wrapf(err, "error retrieving granter for authz transfer")
+			}
+			if authzGranter != "" {
+				granter, err := sdk.AccAddressFromBech32(args[0])
+				if err != nil {
+					return sdkErrors.Wrapf(err, "invalid from address %s", args[0])
+				}
+				transfer := types.NewMsgTransferRequest(granter, from, to, coins[0])
+				msg := authz.NewMsgExec(clientCtx.GetFromAddress(), []sdk.Msg{transfer})
+				return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+			}
 			msg := types.NewMsgTransferRequest(clientCtx.GetFromAddress(), from, to, coins[0])
 			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
 		},
 	}
+	cmd.Flags().String(FlagGranter, "", "When specified the transfer command will be executed on behalf of the granter")
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func GetCmdGrantAuthorization() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "grant-authz [grantee] [authorization_type]",
+		Aliases: []string{"ga"},
+		Args:    cobra.ExactArgs(2),
+		Short:   "Grant authorization to an address",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`grant authorization to an address to execute an authorization type [transfer]:
+
+Examples:
```

Comment:
`Examples` should be its own field in the `cobra.Command` rather than just extra text in the `Long` field.

Reply:
I'm a bit on the fence about this.  I'm trying to keep it consistent with how we are doing it in the current file, where all the other commands have it in the `Long`.  The `metadata` module is the only one that uses the `Example` currently.  Also, the cosmos-sdk uses the `Long` as well.  Here is where I got my command from: https://github.com/cosmos/cosmos-sdk/blob/f961e01bcf19a0f4254864c44306c5126d912e45/x/authz/client/cli/tx.go#L61.  If I modify it, I would want to change all the other commands too, but I would also like to keep this PR to only contain changes for authz support.

Reply:
I made a ticket to clean up the whole project to use the same pattern. #399 This will be a good ticket for new members of the team.

Reply:
Ah. Yeah. I know I've seen other places that include the examples in the `Long` like that. I'm not even really sure when the Examples show up and when they don't. So it probably doesn't even really matter.

I have a dream that one day, I'll go through and standardize all the CLI usage stuff, but eh, that's a lot of work for some cosmetics.

Good to go!